### PR TITLE
Add `AnimatedIcons` diagrams

### PR DESCRIPTION
**Pull request by TahaTesser**
This PR generates all the `AnimatedIcons` for https://github.com/flutter/flutter/issues/113075.

The issue requests images or better, gifs, `diagram_generator` only  support generating `mp4` files for animations.

I could manually run `ffmpeg` command to convert these `.mp4`s to `.gifs` and update this PR.

Like this, `ffmpeg -i add_event.mp4 add_event.gif`  

or 

I'm somewhat familiar with `ffmpeg`, I could add support for generating `gif`s in `bin/generate.dart` in a separate PR.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by TahaTesser** [comment]
Requesting @gspencergoog for review, since he is responsible for adding `ffmpeg` :)

**Comment by gspencergoog** [comment]
We use mp4s and ffmpeg because the files are smaller, and the quality is much better, and we can also do higher frame rates if we want to.  Why would we want to use GIFs?

**Comment by TahaTesser** [comment]
@gspencergoog 
Hans Muller in https://github.com/flutter/flutter/issues/113075 proposed "animated GIFs". 
I can totally the update PR to add tap action to play MP4s. (similar to one found in [SlideTransition](https://master-api.flutter.dev/flutter/widgets/SlideTransition-class.html))

**Comment by TahaTesser** [comment]
@HansMuller 
Do you have any suggestions?

**Comment by gspencergoog** [comment]
These look pretty good, but could you maybe increase the size to a size that is larger than usual so that it's easier to press play, and easier to see the details of the icons?

**Comment by TahaTesser** [comment]
> These look pretty good, but could you maybe increase the size to a size that is larger than usual so that it's easier to press play, and easier to see the details of the icons?

For sure, I only kept them small, in case, we need to convert them to small gifs. For press play, I would increase the size.

**Comment by TahaTesser** [comment]
@gspencergoog 
Updated the size to 72px. 


https://user-images.githubusercontent.com/48603081/195797509-f761675c-8182-4a3b-886f-ec5faeab0dbf.mov

File: packages/diagrams/lib/src/animated_icons.dart [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    super.initState();
+    _controller = AnimationController(
+      duration: _kAnimationDuration,
+      vsync: this,
+    )..addListener(() {
+        setState(() {
+          // The animation controller is changing the animation value, so we
+          // need to redraw.
+        });
+      });
```
Comment (PixelToast) [inline packages/diagrams/lib/src/animated_icons.dart:59]: `AnimatedBuilder` is a cleaner way of rebuilding subtrees when a Listenable updates

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**